File: javascript/client/shop-js/src/models/product.js
export class Product {
    id;
    name;
    category;
    price;
    created_date;
    description;
    available;

    constructor(
        id,
        name,
        category,
        price,
        created_date,
        description,
        available
    ) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.price = price;
        this.created_date = created_date;
        this.description = description;
        this.available = available;
    }
}

File: lua/src/lua/models/product.lua
local Model = require("lapis.db.model").Model

local Product = Model:extend('product', {
    primary_key = "id",
    columns = {
        { "id", type = "serial" },
        { "name", type = "string" },
        { "category", type = "number" },
        { "price", type = "number" },
        { "created_date", type = "date" },
        { "description", type = "string" },
        { "available", type = "boolean" }
    }
})

return Product